valuable/src/structable.rs

```diff
@@ -68,32 +155,134 @@ impl fmt::Debug for dyn Structable + '_ {
 }
 
 impl<'a> StructDef<'a> {
+    /// Create a new `StructDef::Static` instance.
+    ///
```

Comment:
also maybe worth just adding something like
```suggestion
    ///
    /// This should be used when a struct's fields are fixed and known ahead of time.
    ///
```